File: 爬蟲股票投入試算.py
import yfinance as yf
from datetime import datetime

# Ticker symbol
stock_name = 'VOO'
stock = yf.Ticker(stock_name)

# Starting and ending dates
start_date = datetime(2021,1,1)
end_date = datetime(2022,12,31)

# Fetch historical prices
hist = stock.history(start=start_date, end=end_date)

# The amount of regular investment
investment = 1000

# The number of shares bought per investment
shares = 0

# The balance of the investment account
balance = 0

# Track the shares and balance over time
share_track = []
balance_track = []

# Fees in TWD and USD
fee_twd = 600
fee_usd = 25

for date, row in hist.iterrows():
    # Check if date is the first day of the month
    if date.day == 1:
        # Get the exchange rate
        rate = get_exchange_rate(date)
        # Calculate the number of shares bought with the investment after the fees
        shares += (investment-25-600*rate)/row["Open"]
        balance += investment-25-600*rate
        share_track.append(shares)
        balance_track.append(balance)
    # Reinvest dividends
    dividend = row["Dividends"]
    if dividend>0:
        shares += dividend/row["Open"]
        balance += dividend
        
# calculate the final total value
final_value = shares*row["Close"] + balance
print(f'Final total value from {stock_name} from {start_date} to {end_date} is ${final_value}')

File: 爬蟲股票投入試算.swift
import Foundation

let stockName = "VOO"
let startDate = DateComponents(calendar: .current, year: 2015, month: 1, day: 1).date
let endDate = DateComponents(calendar: .current, year: 2022, month: 12, day: 31).date
let investment = 1000.0
var shares = 0.0
var balance = 0.0
let feeTWD = 600.0
let feeUSD = 25.0

// Fetch historical prices
// Code to fetch historical prices goes here

for date in startDate...endDate {
    let dateComponents = Calendar.current.dateComponents([.day, .month, .year], from: date)
    guard let day = dateComponents.day, let month = dateComponents.month, let year = dateComponents.year else {
        continue
    }
    if day == 1 {
        let rate = 1/30
        let price = // Code to get the open price of VOO on this date goes here
        shares += (investment - feeUSD - feeTWD * rate) / price
        balance += investment - feeUSD - feeTWD * rate
    }
    // Code to get dividends goes here
}

let finalValue = shares * // Code to get the closing price of VOO on the last date goes here + balance
print("Final total value from \(stockName) from \(startDate) to \(endDate) is $\(finalValue)")
